ui/js/controllers/perf/compare.js

```diff
@@ -121,63 +123,80 @@ perf.controller('CompareChooserCtrl', [
+                                        newRevision: $scope.newRevision
+                                    });
+                                }
+                                else {
+                                    $state.go('compare', {
+                                        originalProject: $scope.originalProject.name,
+                                        newProject: $scope.newProject.name,
+                                        newRevision: $scope.newRevision,
+                                        selectedTimeRange: compareBaseLineDefaultTimeRange
+                                    });
+                                }
+                            }
+                        });
+                    });
+                // if we have a try push prepopulated, automatically offer a new revision
+                if ($scope.newRevision.length >= 12) {
+                    $scope.updateNewRevisionTips();
+                    $scope.getPreviousRevision();
+                }
             };
-
-            // if we have a try push prepopulated, automatically offer a new revision
-            if ($scope.newRevision.length >= 12) {
-                $scope.updateNewRevisionTips();
-                $scope.getPreviousRevision();
-            }
         });
     }]);
 
 perf.controller('CompareResultsCtrl', [
     '$state', '$stateParams', '$scope', '$rootScope', '$location',
     'thServiceDomain', 'ThRepositoryModel',
     'ThResultSetModel', '$http', '$httpParamSerializer','$q', '$timeout', 'PhFramework', 'PhSeries',
-    'math', 'phTimeRanges', 'PhCompare',
+    'math', 'phTimeRanges', 'PhCompare','compareBaseLineDefaultTimeRange',
     function CompareResultsCtrl($state, $stateParams, $scope,
                                 $rootScope, $location,
                                 thServiceDomain,
                                 ThRepositoryModel, ThResultSetModel, $http, $httpParamSerializer,
                                 $q, $timeout, PhFramework, PhSeries, math,
                                 phTimeRanges,
-                                PhCompare) {
+                                PhCompare, compareBaseLineDefaultTimeRange) {
         function displayResults(rawResultsMap, newRawResultsMap) {
             $scope.compareResults = {};
             $scope.titles = {};
-            window.document.title = ("Comparison between " + $scope.originalRevision +
+            if ($scope.originalRevision !== undefined) {
+                window.document.title = ("Comparison between " + $scope.originalRevision +
                                      " (" + $scope.originalProject.name + ") " +
                                      "and " + $scope.newRevision + " (" + $scope.newProject.name + ")");
+            }
+            else {
+                window.document.title = ("Comparison between " + $scope.originalProject.name +
+                                     " and " + $scope.newRevision + " (" + $scope.newProject.name + ")");
```

Comment:
I think we might as well use ES6 template strings here while we're at it. So change this to:

```js
window.document.title = `Comparison between ${$scope.originalProject.name} and ...`
```

Same with the above.